src/docfx/lib/git/GitUtility.cs

```diff
@@ -202,6 +187,43 @@ public static void CheckMergeConflictMarker(string content, string file)
             }
         }
 
+        /// <summary>
+        /// Clones or update a git repository to the latest version.
+        /// </summary>
+        private static Task CloneOrUpdate(string path, string url, IEnumerable<string> refspecs, bool bare, Config config)
+        {
+            // Unifies clone and fetch using a single flow:
+            // - git init
+            // - git remote set url
+            // - git fetch
+            // - git checkout (if not a bar repo)
+            Directory.CreateDirectory(path);
+
+            if (git_repository_init(out var repo, path, is_bare: bare ? 1 : 0) != 0)
+            {
+                throw new InvalidOperationException($"Cannot initialize a git repo at {path}");
+            }
+
+            if (git_remote_create(out var remote, repo, "origin", url) == 0)
+            {
+                git_remote_free(remote);
+            }
+
+            git_repository_free(repo);
+
+            // Fetch from local file system if we are using a mock repo for testing
+            var mockedRepos = MockedRepos.Value;
+            if (mockedRepos != null && mockedRepos.TryGetValue(url, out var mockedLocation))
+            {
+                url = mockedLocation;
+            }
```

Comment:
can we avoid to mix the test code and prod code, i think we can prepare the NockedRepos before test